### PR TITLE
feat: color and sorting for trades

### LINKED ISSUE #477: highlight actionable trades
- [ ] sort trades so all actionable trades are at the top
- [ ] highlight actionable trades

### PR DESCRIPTION
**Pull request by Nick42791**
Sorting order:
trades sent to you -> color: orange
trades you've sent -> color: orange
trades that don't concern you -> color: white

Additionally, I've added a filter system. The categories are:
Your trades
Trades sent to you
All active trades

should close #477

### CODE REVIEW

File: client/src/components/game/phases/Trades.vue [hunk 1]
```diff
@@ -48,7 +49,26 @@ export default class Trades extends Vue {
       to: tradeSet[id].to,
       status: tradeSet[id].status
     }));
-    return trades;
+    return trades.sort((a,b) => {
```
Comment (chrstngyn) [inline client/src/components/game/phases/Trades.vue:52]: add comments

File: client/src/components/game/phases/trade/ActiveTrade.vue [hunk 2]
```diff
@@ -185,6 +187,14 @@ export default class ActiveTrade extends Vue {
 
   statusColor(type:string){
     let color = 'white';
+
+    if(this.involved == 1){
+      color = 'var(--new-space-orange)';
+    }
+    else if(this.involved == 0){
+      color = 'var(--trade-request-blue)';
```
Comment (chrstngyn) [inline client/src/components/game/phases/trade/ActiveTrade.vue:195]: not sure that having a blue color is necessary. trades could just be orange if they're associated with the player (received or sent by the player)

also any trades colors that are being set should coordinate with mars log message colors, log colors can be changed to match trades or vice versa. but in this case, any color assignments should persist from trades to mars logs. there is a color map for mars log colors in `getters.ts`

**Comment by chrstngyn** [comment]
looks good. merging now

**Comment by alee** [comment]
Two quick notes:

1. Clicking on "Your Trades" or "All Active Trades" briefly brings up all your cancelled trades for some reason
2. the code for activeFilters should be refactored, though I'm glad to see comments!

**Pull request merged**